### PR TITLE
Refactor some more elf code

### PR DESCRIPTION
**Pull request by luc-tielen**
- [x] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

### CODE REVIEW

File: libr/bin/p/bin_elf.inc.c [hunk 1]
```diff
@@ -195,15 +198,14 @@ static void process_constructors(RBinFile *bf, RList *ret, int bits) {
...
 		}
-		if (sec->size > 0xffffff) {
-			continue;
-		}
 		ut8 *buf = calloc (sec->size, 1);
 		if (!buf) {
 			continue;
 		}
+
 		(void)r_buf_read_at (bf->buf, sec->paddr, buf, sec->size);
```
Comment (trufae) [inline libr/bin/p/bin_elf.inc.c:206]: check return to realloc the buffer and maybe add a warning if that section->size needs to be trimmed to fit the reality

File: libr/bin/p/bin_elf.inc.c [hunk 2]
```diff
@@ -195,15 +198,14 @@ static void process_constructors(RBinFile *bf, RList *ret, int bits) {
 		if (type == -1) {
 			continue;
 		}
-		if (sec->size > 0xffffff) {
```
Comment (trufae) [inline libr/bin/p/bin_elf.inc.c:198]: this is done to avoid huge allocations. which is an OOM crash in the fuzzing loop

Comment (luc-tielen) [reply libr/bin/p/bin_elf.inc.c:198]: I figured, but I only moved it up to save some work if a section is too large

File: libr/bin/p/bin_elf.inc.c (hunk 1 above)
Comment (luc-tielen) [reply libr/bin/p/bin_elf.inc.c:206]: can you be more clear what needs to happen here?
```
int size = r_buf_read_at(...)
if (size != sec->size) {
        if (size < sec->size) { R_LOG_WARN ("unexpected section size"); }
        realloc (buf, size);
        sec->size = size;
}
```
something like this?

File: libr/bin/p/bin_elf.inc.c [hunk 3]
```diff
@@ -179,10 +179,13 @@ static void process_constructors(RBinFile *bf, RList *ret, int bits) {
 	RList *secs = sections (bf);
 	RListIter *iter;
 	RBinSection *sec;
-	int i, type;
 	r_list_foreach (secs, iter, sec) {
-		type = -1;
+		if (sec->size > 0xffffff) {
```
[the remark this answers is not in the document]
Comment (luc-tielen) [reply libr/bin/p/bin_elf.inc.c:183]: good point!

File: libr/bin/p/bin_elf.inc.c (hunk 1 above)
Comment (trufae) [reply libr/bin/p/bin_elf.inc.c:206]: Yep like this

**Pull request merged**